### DISCUSSION TITLE
Enable the language server without mason installed binaries

### DISCUSSION THREAD

**Discussion by s1n7ax** [LazyVim/LazyVim#2325] [Q&A] [answered]
I'm using NixOS and most of the binaries installed by mason cannot be run on NixOS. Few of them are installed by default template

- Stylua
- Lua_ls

My requirement is to enable language server setup however without the binary from mason. Does anyone know if this is possible?

**Comment by dpetka2001** [comment, marked as answer]
```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        lua_ls = {
          mason = false,
        },
      },
    },
  },
}
```
The server won't be installed by Mason and instead should use the binary that is installed on your system and is available in your $PATH.

**Reply by s1n7ax** [reply]
Noice.

**Comment by s1n7ax** [comment]
@dpetka2001 Is there something similar to disable the `stylua` installation?

**Comment by s1n7ax** [comment]
Following worked for `stylua` & `shfmt`

```lua
return {
	'williamboman/mason.nvim',
	opts = {
		ensure_installed = {},
	},
}
```

**Reply by dpetka2001** [reply]
@s1n7ax Please be aware that if you have your top level mason `ensure_installed` like this, other defaults will be overwritten if you want to later include some LazyVim extras. The correct way would be to use `vim.list_extend()` to extend the default implementation (i guess "correct" is a bit strong and subjective to what a user wants to do in the end), but if it's just for your personal use and know that you will have to maintain your own Mason `ensure_installed`, then I guess there's no harm as long as you're aware.

**Reply by s1n7ax** [reply]
Ah. I see

**Reply by s1n7ax** [reply]
Updated accordingly.

```lua
return {
	'williamboman/mason.nvim',
	opts = function(_, opts)
		opts.ensure_installed = vim.tbl_filter(function(value)
			return value == 'stylua' or value == 'shfmt'
		end, opts.ensure_installed)

		return opts
	end,
}
```

**Reply by dpetka2001** [reply]
Maybe instead of `return value == 'stylua' or value == 'shfmt'` you need `return not vim.tbl_contains({ "stylua", "shfmt" }, value)`? You want to have an empty `ensure_installed` for mason to not install them if I understood correctly?
